### PR TITLE
clickhouse http api support x-clickhouse-user.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #issue

### CODE REVIEW

**Review by junnplus — APPROVED** [review]

File: query/src/servers/http/middleware.rs [hunk 1]
```diff
@@ -30,30 +30,46 @@ use poem::Middleware;
...
-    }
     let client_ip = match req.remote_addr().0 {
         Addr::SocketAddr(addr) => Some(addr.ip().to_string()),
         Addr::Custom(..) => Some("127.0.0.1".to_string()),
         _ => None,
     };
+    if auth_headers.is_empty() {
+        if let HttpHandlerKind::Clickhouse = kind {
+            let (user, key) = (
+                req.headers().get("X-CLICKHOUSE-USER"),
```
Comment (Xuanwo) [inline query/src/servers/http/middleware.rs:57]: Is the naming used by clickhouse? How about using `X-CLICKHOUSE-USERNAME` and `X-CLICKHOUSE-PASSWORD` to match with HTTP basic auth?

Comment (youngsofun) [reply query/src/servers/http/middleware.rs:57]: clickhouse just use it, we need to be compatible

**Comment by youngsofun** [comment]
fixed the flaky test in this pr

@sundy-li @ZeaLoVe @leiysky @junnplus @Xuanwo @dantengsky 

I'm awfully sorry

**Review by Xuanwo — APPROVED** [review]

**Comment by youngsofun** [comment]
@zhang2014  is this a flaky test, seems not related to this pr

```

20_0003_crash_me_table:                                                 [ FAIL ] - result differs with:
--- /workspace/tests/suites/0_stateless/20+_others/20_0003_crash_me_table.result	2022-06-10 01:36:33.043356360 +0000
+++ /workspace/tests/suites/0_stateless/20+_others/20_0003_crash_me_table.stdout	2022-06-10 01:40:14.638532413 +0000
@@ -0,0 +1 @@
+ERROR 1105 (HY000) at line 1: Code: 1054, displayText = Expected server error code: 1104 but got: Ok..
```

**Comment by youngsofun** [comment]
@Xuanwo  
`Queue: Embarked in merge train ` failed
why 
The pull request embarked with main (9c713b1) cannot be merged and has been disembarked

**Comment by Xuanwo** [comment]
> [](/apps/github-actions) ** Developing / test_stateless_standalone_linux (pull_request) **

Because `Developing / test_stateless_standalone_linux (pull_request)` failed.

**Pull request merged**